api/src/main/java/org/openmrs/module/fhir2/providers/r3/EncounterFhirResourceProvider.java

```diff
@@ -152,4 +160,69 @@ public IBundleProvider searchEncounter(@OptionalParam(name = Encounter.SP_DATE)
 		                encounterType, tag, hasAndListParam, id, lastUpdated, sort, includes, revIncludes)));
 	}
 	
+	/**
+	 * Custom search endpoint that fetches encounters that include medication requests Returns a bundle
+	 * that includes the medication requests and any medication dispenses that reference those requests
+	 *
+	 * @param date only return encounters with an encounter date greater than or equal to this date
+	 * @param status if set to active, when determined encounters to include, exclude encounters that
+	 *            *only* have completed or cancelled medication requests
+	 * @param patientSearchTerm restrict to encounters for patietns who name or identifier matches the
+	 *            search term
+	 * @return
+	 */
+	
+	@Search(queryName = "encountersWithMedicationRequests")
```

Comment:
One thing to be careful with: FHIR already has some logic for implementing dates as part of the search to specify ranges ([docs](https://www.hl7.org/fhir/search.html#date)). I think HAPI's implementation will already handle parsing the date with prefix, but I don't exactly know what will happen if you feed it a raw date... Since this is custom, it's probably ok to do whatever, but good to be aware of.

Reply:
Yes, I went back and forth on this... I went with it being opinionated and taking only a raw date, and then explicity setting it as a greater than query a few lines further down, rather than having to rely on the consumer specified the ge flag (since less or equals doesn't really make much sense, currently, for this query)